flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -473,6 +412,62 @@ public WriteOptions getWriteOptions() {
 		return sharedRocksKeyBuilder;
 	}
 
+	public RocksDB getDb() {
+		return db;
+	}
+
+	/**
+	 * Allow using mocks in unit test.
+	 */
+	@VisibleForTesting
+	public void setDb(RocksDB db) {
+		this.db = db;
+	}
+
+	public RocksDBWriteBatchWrapper getWriteBatchWrapper() {
```

Comment:
Will remove most newly added ones and only left `getCheckpointSnapshotStrategy` and `getCheckpointSnapshotStrategy` which is required by `RocksDBStateBackendTest` which seems  hard to rewrite.

Reply:
Please take a look at my later comments, I think the test is really a good candidate for rewrite because it can benefit now from the refactoring, because we can now supply the spy instance of RocksDB through the builder or constructor and don't have to use `RocksDBStateBackend` to get an instance.